src/client/client.c

```diff
@@ -161,6 +162,129 @@ void unit_list_unref(UnitList *unit_list) {
         free(unit_list);
 }
 
+static sd_bus_message *wait_for_job(Client *client, char *object_path) {
+        int r = 0;
+
+        assert(client->waiting_for_job == NULL);
+        client->waiting_for_job = object_path;
+
+        for (;;) {
+                /* Process requests */
+                r = sd_bus_process(client->api_bus, NULL);
+                if (r < 0) {
+                        fprintf(stderr, "Failed to process bus: %s\n", strerror(-r));
+                        return NULL;
+                }
+
+                /* Did we get the result? */
+                if (client->waiting_for_job_result != NULL) {
+                        return steal_pointer(&client->waiting_for_job_result);
+                }
+
+                if (r > 0) { /* request processed, try to process another one */
```

Comment:
Shouldn't this `r > 0` check be right below the `r < 0` check?

Reply:
Ah ok, so the `waiting_for_job_result` is the "successful anchor" for this loop.  And if no process was made `(r == 0)` we wait for the next request. It would be a bit clearer to me if this check would be `r==0` and wait then - it would get rid of the `continue`, but on the other side add another nesting level.

Reply:
I don't see anything wrong with `continue`. We check all conditions to break the loop, and if nothing is relevant - we proceed to waiting for the next event.